### PR TITLE
hm_remove()

### PR DESCRIPTION
**Pull request by harou24**

### CODE REVIEW

File: tests/src/hashmap.cpp [hunk 1]
```diff
@@ -63,3 +63,23 @@ TEST_CASE("hm_set further testing", "[hashmap]")
 	hm_destroy(hashmap, free);
 }
 
+TEST_CASE("hm_remove test", "[hashmap]")
+{
+	void *hashmap = hm_new(HM_SIZE);
```
Comment (s-t-a-n) [inline tests/src/hashmap.cpp:68]: best to not use HM_SIZE but use a for loop to test from size 1 to something like 1024 or something.

you can just put the whole test block in the for loop.

that way you test for more than one size (you could do this for every unittest)

**Review by s-t-a-n — COMMENTED** [review]
very nice man! just a few optimization/style things. but it's already very readable code

**Comment by harou24** [comment]
@s-t-a-n thanks for the review, it helps to improve and write clean and optimized code :)

File: src/hash_map.c [hunk 2]
```diff
@@ -109,6 +109,63 @@ void			hm_destroy(void *_hm, void (*f)(void *))
...
+	t_hash_map	*hm;
+	t_hm_node	**to_remove;
+	
+	hm = (t_hash_map *)_hm;
+	to_remove = &hm->nodes[hash(hm->cap, _key)];
+	if (*to_remove)
+	{
+		if (ft_strcmp((*to_remove)->key, _key) != 0)
+		{
+			*to_remove = find_node_by_key(_hm, _key);
```
Comment (s-t-a-n) [inline src/hash_map.c:160]: add argument to find_node_by_key : node, and start iterating over node->next starting on that note

**Pull request merged**